File: javascript/easy/maximumAverageSubarrayI.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var findMaxAverage = function(nums, k) {
    let maxAverage = 0, sum = 0, n = Math.min(nums.length, k);
    for(let i = 0; i < n; i++) {
        sum += nums[i];
    }
    maxAverage = (sum/n);
    for(let i = n; i < nums.length; i++) {
        sum = sum - nums[i-n] + nums[i];
        maxAverage = Math.max(maxAverage, (sum/n));
    }
    return maxAverage;
};

File: ruby/easy/maximum_average_subarray_i.rb
# @param {Integer[]} nums
# @param {Integer} k
# @return {Float}
def find_max_average(nums, k)
    max_average = sum = 0
    n = [nums.length, k].min
    for i in 0..n-1
        sum += nums[i]
    end
    max_average = (sum.to_f/n)
    for i in n..nums.length-1
        sum = sum - nums[i-n] + nums[i]
        max_average = [max_average, (sum.to_f/k)].max
    end
    return max_average
end
